### PR TITLE
fix: compare RSA blocks instead of parsing

### PR DESCRIPTION
**Pull request by ejohnstown**
RFC 8332 section 3 advises against verifying RSASSA-PKCS1-v1_5 by applying the key and parsing the output. wolfSSH_RsaVerify() builds the expected EMSA-PKCS1-v1_5 block, applies the key with a raw public operation, and compares whole modulus-sized blocks.

- verify through wc_RsaFunction() and a constant-time block compare
- reject a key too small for the digest and minimum padding
- reject a wc_RsaEncryptSize() error rather than casting it
- carve the three working blocks from one allocation
- add test_RsaVerify_BadPadding() for pad and block-type errors

The test passes against the old code too; wolfSSL's unpadding is already strict, so it guards the property rather than a live defect.

Issue: F-10574

### CODE REVIEW

File: tests/unit.c [hunk 1]
```diff
@@ -9394,6 +9394,135 @@ static int test_RsaVerify_BadDigest(void)
...
+
+    ret = wc_Hash(WC_HASH_TYPE_SHA256, data, sizeof(data),
+            digest, WC_SHA256_DIGEST_SIZE);
+    if (ret != 0) { result = -524; goto done; }
+
+    encDigestSz = wc_EncodeSignature(encDigest, digest,
+            WC_SHA256_DIGEST_SIZE, wc_HashGetOID(WC_HASH_TYPE_SHA256));
+    if (encDigestSz <= 0) { result = -525; goto done; }
+
+    padSz = keySz - (word32)encDigestSz - 3;
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply tests/unit.c:9459]: Agreed, added the RSA_MIN_PAD_SZ guard before computing padSz.

File: src/internal.c [hunk 2]
```diff
@@ -14684,35 +14684,45 @@ int wolfSSH_RsaVerify(const byte *sig, word32 sigSz,
...
-#ifdef WOLFSSH_SMALL_STACK
+
+    /* One block each for the normalized signature, the recovered value, and
+     * the expected encoding. */
     if (ret == WS_SUCCESS) {
-        checkDigest = (byte*)WMALLOC(MAX_ENCODED_SIG_SZ, heap, DYNTYPE_TEMP);
-        if (checkDigest == NULL)
+        scratch = (byte*)WMALLOC(keySz * 3, heap, DYNTYPE_TEMP);
+        if (scratch == NULL)
             ret = WS_MEMORY_E;
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply src/internal.c:14744]: Not reachable. keySz is wc_RsaEncryptSize() of a decoded key, bounded by wolfSSL RSA_MAX_SIZE: 512 bytes at the 4096-bit default, 2048 at the 16384-bit ceiling. Overflowing a word32 would need a modulus near a gigabyte, which no SSH packet can carry.

File: tests/unit.c [hunk 3]
```diff
@@ -9394,6 +9394,270 @@ static int test_RsaVerify_BadDigest(void)
...
+               ret, WS_RSA_E);
+        result = -550;
+        goto done;
+    }
+
+    /* A digest the key has no room to pad is refused before the padding size
+     * underflows. */
+    WMEMSET(bigDigest, 0, sizeof(bigDigest));
+    ret = wolfSSH_TestRsaVerify(sig, sigSz, bigDigest,
+            keySz - RSA_MIN_PAD_SZ + 1, &key, NULL);
```
Comment (ejohnstown) [inline tests/unit.c:9631]: Confirmed. I stubbed the guard out and rebuilt: the test still reported SUCCESS, so it was asserting nothing about the guard.

Kept the boundary case, since it documents the smallest digest the guard rejects, and added a second call with encDigestSz of keySz - 2. That is past keySz - 3, so without the guard padSz underflows to 0xFFFFFFFF and the pad WMEMSET runs off the scratch allocation. With the guard stubbed out the test now dies on a signal; with it in place the test passes. bigDigest is 512 bytes, so the larger read stays in bounds.

**Review by philljj — APPROVED** [review]
